src/miner.cpp

```diff
@@ -148,12 +147,14 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
     coinbaseTx.vout.resize(1);
     coinbaseTx.vout[0].scriptPubKey = scriptPubKeyIn;
     coinbaseTx.vout[0].nValue = nFees + GetBlockSubsidy(nHeight, chainparams.GetConsensus());
-    coinbaseTx.vin[0].scriptSig = CScript() << nHeight << OP_0;
+    coinbaseTx.vin[0].scriptSig = CScript() << nHeight << CScriptNum(1);
```

Comment:
The extraNonce is incremented before any block is generated, which adds 4 bytes to the block weight. This adds an extraNonce of 1 here so that the following `GetBlockWeight` calculation is accurate and not 4 bytes less.